File: src/main/java/tn/esprit/dari3d/Entities/IngenieurDetail.java
package tn.esprit.dari3d.Entities;

import jakarta.persistence.*;
import lombok.Data;

import java.util.Set;

@Entity
@Data
public class IngenieurDetail {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String experience;
    private String description;

    @OneToOne
    @JoinColumn(name = "user_id")
    private Users user;

    @OneToMany(mappedBy = "ingenieurDetail")
    private Set<PicturesOfDoneProjects> picturesOfDoneProjects;


}

File: frontend/src/app/core/models/IngenieurDetail.ts
import { PicturesOfDoneProjects } from "./PicturesOfDoneProjects";
import { Users } from "./Users";

export class IngenieurDetail {
    id: number;
    Experience: string;
    Description:string;
    user:Users;
    pictures:PicturesOfDoneProjects[];
}
